devices/client.go

```diff
@@ -0,0 +1,57 @@
+package devices
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/tidepool-org/devices/api"
+	"go.uber.org/fx"
+	"google.golang.org/grpc"
+
+	"github.com/tidepool-org/platform/errors"
+)
+
+var ClientModule = fx.Provide(NewClient)
+
+type Config struct {
+	DevicesClientAddress string `envconfig:"TIDEPOOL_DEVICES_CLIENT_ADDRESS" required:"true"`
+}
+
+type Client struct {
+	api.DevicesClient
+	conn *grpc.ClientConn
+}
+
+func NewClient(lifecycle fx.Lifecycle) api.DevicesClient {
+	client := &Client{}
+	lifecycle.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			return client.Initialize(ctx)
+		},
+		OnStop: func(ctx context.Context) error {
+			return client.Stop()
+		},
+	})
+
+	return client
+}
+
+func (c *Client) Initialize(ctx context.Context) error {
+	cfg := &Config{}
+	if err := envconfig.Process("", cfg); err != nil {
+		return err
+	}
+
+	conn, err := grpc.DialContext(ctx, cfg.DevicesClientAddress, grpc.WithInsecure(), grpc.WithBlock())
```

Comment:
Is `WithInsecure` okay? I assume that encryption is handled by mTLS in linkerd/gloo, but just wanted to be sure.

Reply:
It's ok - mTLS is indeed handled by linkerd.